GameServerLib/GameObjects/AttackableUnits/AI/ObjAIBase.cs

```diff
@@ -230,17 +230,24 @@ public void RemoveBuffSlot(Buff b)
 
         public void ApplyCrowdControl(UnitCrowdControl cc)
         {
-            if (cc.IsTypeOf(CrowdControlType.STUN) || cc.IsTypeOf(CrowdControlType.ROOT))
+            if (cc.IsTypeOf(CrowdControlType.ROOT))
             {
-                StopMovement();
+                Stats.SetActionState(ActionState.CAN_MOVE,false);
```

Comment:
How about creating a function
```CS
public void LockMovement()
{
    Stats.SetActionState(ActionState.CAN_MOVE, false);
    Stats.SetActionState(ActionState.CAN_NOT_MOVE, true);
    StopMovement();
}
```
And calling it here?

Also add back the stun check, or an additional if check for the stun that does the same thing, however you see fit better.